File: 2021_3/day6-Short Encoding of Words/python3/Short Encoding of Words.py
class Solution:
    def minimumLengthEncoding(self, words: List[str]) -> int:
        indices = set(words)
        
        for word in words:
            for i in range(1,len(word)):
                indices.discard(word[i:])
              
        ans = 0
        for word in indices:
            ans += len(word) + 1
        return ans

File: 2021_3/day6-Short Encoding of Words/typescript/Short Encoding of Words.ts
function minimumLengthEncoding(words: string[]): number {
    let dict = new Set(words);
    
    for(let word of words) {
        for(let i = 1; i<word.length; i++) {
            dict.delete(word.substring(i));
        }
    }
    
    let ans : number = 0;
    for(let word of dict) 
        ans += word.length + 1
    
    return ans;
};